.github/scripts/check-header.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+
+import os, sys
+from argparse import ArgumentParser
+from difflib import unified_diff
+from json import load
+
+def dprint(msg):
+    print('[DEBUG]: %s' % str(msg))
+
+class HeaderChecker:
+    def __init__(self, header, padding=1000):
+        self.padding = padding
+        self.header = header
+
+    def normalizeHeader():
+        assert False, 'Unimplemented'
+
+    def checkJSONList(self, path_json):
+        '''
+        This is particularly useful when ingesting output from other programs, like git actions
+        '''
+        assert os.path.exists(path_json), 'No such file: ' + path_json
+
+        # Get list of files to check from JSON file
+        with open(path_json) as file_json:
+            file_checklist = load(file_json)
+            assert isinstance(file_checklist, list), 'Expected list for singular JSON List entry'
+
+        # Accrue how how files fail the check
+        n_failed = 0
+        for path_file in file_checklist:
+            assert isinstance(path_file, str), 'Unexpected JSON format for ' + path_json
+            n_failed += not self.isValidFile(path_file)
+
+        return n_failed
+
+    def isValidFile(self, path):
+        assert os.path.exists(path), 'No such file: ' + path
+
+        # Don't need entire file. Read sufficienly large chunk of file that should contain the header
+        with open(path, encoding='utf-8', errors='ignore') as file:
+            chunk = file.read(len(''.join(self.header)) + self.padding)
+            lines = [('%s\n' % l) for l in chunk.strip().splitlines()][:len(self.header)]
+            if self.header == lines:
+                return True
+            else:
+                print('File Delta: %s' % path)
+                print(*unified_diff(lines[:len(self.header)], self.header))
+                return False
+
+
+def configArgParser():
+    parser = ArgumentParser(description='FreeRTOS file header checker. We expect a consistent header across all '
+                                        'first party files. The header includes current version number, copyright, '
+                                        'and FreeRTOS license.')
```

Comment:
Are we also checking the 2 links?

Reply:
Yes, for now the expected `freertos_header` and `kernel_header` are set in main and include version number, copyright, license, and the two links. If anything diffs here, it reports a diff similar looking to gitdiff. Additionally any diffs in any of the checked files will fail the git action